alice/middleware/logger.go

```diff
@@ -22,35 +27,30 @@ type logger struct {
 	logRequests bool
 }
 
-type loggingResponseWriter struct {
-	http.ResponseWriter
-	StatusCode int
-}
-
-func (l *loggingResponseWriter) WriteHeader(code int) {
-	l.StatusCode = code
-	l.ResponseWriter.WriteHeader(code)
-}
-
 // NewLogrusLogger allows you to setup a base entry to use for logging
 func NewLogrusLogger(baseEntry *logrus.Entry, logRequests bool) Logger {
 	return &logger{entry: baseEntry, logRequests: logRequests}
 }
 
 func (l *logger) Log(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		loggingResponseWriter := &lrw.LoggingResponseWriter{w, http.StatusOK, nil}
+		h.ServeHTTP(loggingResponseWriter, r)
 		entry := l.entry.WithFields(logrus.Fields{
-			logFieldRequestID: getRequestIDFromContext(r.Context()),
-			logFieldUserID:    getInsecureUserIDFromContext(r.Context()),
+			logFieldRequestID:  getRequestIDFromContext(r.Context()),
+			logFieldUserID:     getInsecureUserIDFromContext(r.Context()),
+			logFieldMethod:     r.Method,
+			logFieldStatusCode: loggingResponseWriter.StatusCode,
+			logFieldPath:       r.URL.String(),
+			logFieldError:      loggingResponseWriter.InnerError,
 		})
+
 		// add logger to the context
 		ctx := context.WithValue(r.Context(), contextKeyLogger, entry)
```

Comment:
all the logging in the endpoints use the logger that is in this context.  we'd only want to add some of these if there is an error